### PR TITLE
fix: replace web auth with casdoor-js-sdk

### LINKED ISSUE #338: Use casdoor-js-sdk to replace the code inside the repo
Use: https://github.com/casdoor/casdoor-js-sdk to replace: https://github.com/casbin/casnode/tree/master/web/src/auth

### PR DESCRIPTION
**Pull request by WindSpiritSR**
Fix: https://github.com/casbin/casnode/issues/338

Deployed at: https://forum.leviatan.cn/

Signed-off-by: WindSpiritSR <simon343riley@gmail.com>

### CODE REVIEW

**Comment by hsluoyz** [comment]
@kocoler @Kininaru

**Review by kocoler — APPROVED** [review]

**Review by hsluoyz — CHANGES_REQUESTED** [review]
@WindSpiritSR You should replace all the Auth files with SDK. By only replacing half of one file, I think the casdoor-js-sdk is then useless.

image

File: web/src/Conf.js [hunk 1]
```diff
@@ -15,6 +15,8 @@
 export const AuthConfig = {
   // serverUrl: "https://door.casbin.com",
   serverUrl: "http://localhost:7001",
+  // your Casnode backend URL
+  casnodeUrl: "http://localhost:7000",
```
Comment (hsluoyz) [inline web/src/Conf.js]: `casnodeUrl` is used nowhere.

File: web/src/main/FilesBox.js [hunk 2]
```diff
@@ -15,8 +15,8 @@
 import React from "react";
 import * as FileBackend from "../backend/FileBackend";
 import * as Setting from "../Setting";
```
Comment (hsluoyz) [inline web/src/main/FilesBox.js:17]: Setting is already imported

File: web/src/main/NewBox.js [hunk 3]
```diff
@@ -18,8 +18,8 @@ import * as NodeBackend from "../backend/NodeBackend";
 import * as MemberBackend from "../backend/MemberBackend";
 import * as TopicBackend from "../backend/TopicBackend";
 import * as Setting from "../Setting";
```
Comment (hsluoyz) [inline web/src/main/NewBox.js:20]: Setting is already imported

File: web/src/main/NodeBox.js [hunk 4]
```diff
@@ -14,6 +14,7 @@
 
 import React from "react";
 import * as Setting from "../Setting";
```
Comment (hsluoyz) [inline web/src/main/NodeBox.js]: Setting is already imported

File: web/src/main/SignoutBox.js [hunk 5]
```diff
@@ -16,8 +16,9 @@ import React from "react";
 import Header from "./Header";
 import { withRouter } from "react-router-dom";
 import * as AccountBackend from "../backend/AccountBackend";
-import * as Auth from "../auth/Auth";
 import i18next from "i18next";
+import * as Setting from "../Setting";
```
Comment (hsluoyz) [inline web/src/main/SignoutBox.js:20]: Setting is already imported

File: web/src/rightbar/RightSigninBox.js [hunk 6]
```diff
@@ -14,9 +14,10 @@
 
 import React from "react";
 import { withRouter } from "react-router-dom";
-import * as Auth from "../auth/Auth";
 import * as Conf from "../Conf";
 import i18next from "i18next";
+import * as Setting from "../Setting";
```
Comment (hsluoyz) [inline web/src/rightbar/RightSigninBox.js:19]: Setting is already imported

File: web/src/Header.js [hunk 7]
```diff
@@ -19,9 +19,7 @@ import * as Setting from "./Setting";
 import * as Conf from "./Conf";
```
Comment (hsluoyz) [inline web/src/Header.js:19]: Setting is already imported

File: web/src/AuthCallback.js [hunk 8]
```diff
@@ -15,8 +15,8 @@
 import React from "react";
 import { Button, Result, Spin } from "antd";
 import { withRouter } from "react-router-dom";
-import * as AccountBackend from "../backend/AccountBackend";
-import * as Util from "./Util";
+import * as Setting from "./Setting";
```
Comment (hsluoyz) [inline web/src/AuthCallback.js:18]: Setting is already imported

File: web/package.json [hunk 9]
```diff
@@ -11,6 +11,7 @@
     "antd": "^4.16.8",
     "braft-editor": "^2.3.9",
     "browser-image-size": "^1.1.0",
+    "casdoor-js-sdk": "0.0.1",
```
Comment (hsluoyz) [inline web/package.json]: yarn.lock?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: web/yarn.lock [hunk 10]
```diff
@@ -3064,6 +3064,11 @@ capture-exit@^2.0.0:
   dependencies:
     rsvp "^4.8.4"
 
+casdoor-js-sdk@0.0.2:
+  version "0.0.2"
+  resolved "https://registry.nlark.com/casdoor-js-sdk/download/casdoor-js-sdk-0.0.2.tgz#253cea3f7316f9cbb3fbda01d2bf0b94c3fc8508"
```
Comment (hsluoyz) [inline web/yarn.lock]: Use: `registry.yarnpkg.com`

File: web/src/Setting.js [hunk 11]
```diff
@@ -45,6 +47,30 @@ export function initFullClientUrl() {
...
 
+export function initCasdoorSdk(config) {
+  CasdoorSdk = new Sdk(config);
+}
+
+export function getSignupUrl() {
+  return CasdoorSdk.getSignupUrl();
+}
+
+export function getSigninUrl(redirectUrl) {
```
Comment (hsluoyz) [inline web/src/Setting.js]: No arg for `getSigninUrl()`

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**